File: 228/summary-ranges.go
package main

import "fmt"

func summaryRanges(nums []int) []string {
	var res []string

	n := len(nums)
	if n == 0 {
		return res
	}

	begin := nums[0]
	str := ""

	for i := 0; i < n; i++ {
		if i == n-1 || nums[i]+1 != nums[i+1] {
			if nums[i] == begin {
				str = fmt.Sprintf("%d", begin)
			} else {
				str = fmt.Sprintf("%d->%d", begin, nums[i])
			}

			if i+1 < n {
				begin = nums[i+1]
			}

			res = append(res, str)
		}
	}

	return res
}

func main() {
	nums := []int{0, 1, 2, 4, 5, 7}
	fmt.Println(summaryRanges(nums))
}

File: 228/summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function (nums) {
    let res = [];
    let n = nums.length;

    if (!n) return res;

    let begin = nums[0];
    let str;

    for (let i = 0; i < n; i++) {
        if (i == n-1 || nums[i]+1 != nums[i+1]) {
            if (nums[i] == begin) {
                str = `${begin}`;
            } else {
                str = `${begin}->${nums[i]}`;
            }

            if (i+1 < n) {
                begin = nums[i+1];
            }

            res.push(str);
        }
    }

    return res;
};